Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1637,37 +1648,42 @@ open class NavigationMapView: UIView {
                                                   name: .navigationCameraStateDidChange,
                                                   object: nil)
     }
-    
-    func setupMapView(_ frame: CGRect,
-                      navigationCameraType: NavigationCameraType = .mobile,
-                      tileStoreLocation: TileStoreConfiguration.Location? = .default) {
+
+    private func makeMapView(
+            frame: CGRect,
+            navigationCameraType: NavigationCameraType = .mobile,
+            tileStoreLocation: TileStoreConfiguration.Location? = .default
+    ) -> MapView {
         let accessToken = ResourceOptionsManager.default.resourceOptions.accessToken
-        
+
         // TODO: allow customising tile store location.
         let tileStore = tileStoreLocation?.tileStore
-        
+
         // In case of CarPlay, use `pixelRatio` value, which is used on second `UIScreen`.
         var pixelRatio = UIScreen.main.scale
         if navigationCameraType == .carPlay, UIScreen.screens.indices.contains(1) {
             pixelRatio = UIScreen.screens[1].scale
         }
-        
+
         let mapOptions = MapOptions(constrainMode: .widthAndHeight,
```

Comment:
Would it be possible to expose `MapOptions` publicly? I remember there were issues with that reported in https://github.com/mapbox/mapbox-navigation-ios/issues/3793. Maybe situation has changed though.